src/edges_cal/reflection_coefficient.py

```diff
@@ -31,162 +50,472 @@ def gamma2impedance(
+    s22
+        The S22 of the two-port network for the port facing the device under test (DUT)
+    gamma_ref
+        The reflection coefficient of the device
+        under test (DUT) measured at the reference plane.
+
+    Returns
+    -------
+    gamma
+        The intrinsic reflection coefficient of the DUT.
+
+    See Also
+    --------
+    gamma_embed
+        The inverse function to this one.
+    """
+    return (gamma_ref - s11) / (s22 * (gamma_ref - s11) + s12s21)
+
+
+def gamma_embed(
+    s11: np.typing.ArrayLike,
+    s12s21: np.typing.ArrayLike,
+    s22: np.typing.ArrayLike,
+    gamma: np.typing.ArrayLike,
+) -> np.typing.ArrayLike:
+    """Obtain the intrinsic reflection coefficient.
+
+    See Eq. 1 of Monsalve et al., 2016.
 
+    Obtains the instrinsic reflection coefficient from the
+    one measured at the reference plane, given a set of
+    reflection coefficients.
 
-def gamma_shifted(s11, s12s21, s22, r):  # noqa
-    return s11 + (s12s21 * r / (1 - s22 * r))
+    Parameters
+    ----------
+    s11
+        The S11 parameter of the two-port network for the port facing the calibration
+        plane.
+    s12s21
+        The product of ``S12*S21`` of the two-port
+        network.
+    s22
+        The S22 of the two-port network for the port facing the device under test (DUT)
+    gamma
+        The intrinsic reflection coefficient of the device
+        under test (DUT);.
 
+    Returns
+    -------
+    gamma_ref
+         The reflection coefficient of the DUT
+         measured at the reference plane.
+
+    See Also
+    --------
+    gamma_de_embed
+        The inverse function to this one.
+    """
+    return s11 + (s12s21 * gamma / (1 - s22 * gamma))
+
+
+def de_embed(
+    gamma_open_intr: np.ndarray | float,
+    gamma_short_intr: np.ndarray | float,
+    gamma_match_intr: np.ndarray | float,
+    gamma_open_meas: np.ndarray,
+    gamma_short_meas: np.ndarray,
+    gamma_match_meas: np.ndarray,
+    gamma_ref,
+) -> tuple[np.ndarray, np.ndarray, np.ndarray, np.ndarray]:
+    """Obtain network S-parameters from OSL standards and intrinsic reflections of DUT.
+
+    See Eq. 3 of Monsalve et al., 2016.
+
+    Parameters
+    ----------
+    gamma_open_intr
+        The intrinsic reflection of the open standard
+        (assumed as true) as a function of frequency.
+    gamma_shrt_intr
+        The intrinsic reflection of the short standard
+        (assumed as true) as a function of frequency.
+    gamma_load_intr
+        The intrinsic reflection of the load standard
+        (assumed as true) as a function of frequency.
+    gamma_open_meas
+        The reflection of the open standard
+        measured at port 1 as a function of frequency.
+    gamma_shrt_meas
+        The reflection of the short standard
+        measured at port 1 as a function of frequency.
+    gamma_load_meas
+        The reflection of the load standard
+        measured at port 1 as a function of frequency.
+    gamma_ref
```

Comment:
@steven-murray I thought here: gamma_xx_measured and gamma_ref are functions of frequency measured at the same point. For gamma_xx_meas we say port 1 and for gamma_ref we say reference plane

Reply:
Hmmm, I'm not sure exactly what you mean?